addons/l10n_ec_website_sale/models/payment_method.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class PaymentMethod(models.Model):
+    _inherit = 'payment.method'
+
+    l10n_ec_sri_payment_id = fields.Many2one(
+        comodel_name="l10n_ec.sri.payment",
+        string="SRI Payment Method",
+    )
+
+    fiscal_country_codes = fields.Char(compute="_compute_fiscal_country_codes")
+
+    @api.depends_context('allowed_company_ids')
+    def _compute_fiscal_country_codes(self):
+        for record in self:
+            record.fiscal_country_codes = ",".join(self.env.companies.mapped('account_fiscal_country_id.code'))
+
+    def _get_compatible_payment_methods(
+        self, provider_ids, partner_id, currency_id=None, force_tokenization=False,
+        is_express_checkout=False
+    ):
+        """ Search and return the payment methods matching the compatibility criteria.
+
+        The compatibility criteria are that payment methods must: be supported by at least one of
+        the providers; support the country of the partner if it exists; be primary payment methods
+        (not a brand). If provided, the optional keyword arguments further refine the criteria.
+
+        :param list provider_ids: The list of providers by which the payment methods must be at
+                                  least partially supported to be considered compatible, as a list
+                                  of `payment.provider` ids.
+        :param int partner_id: The partner making the payment, as a `res.partner` id.
+        :param int currency_id: The payment currency, if known beforehand, as a `res.currency` id.
+        :param bool force_tokenization: Whether only payment methods supporting tokenization can be
+                                        matched.
+        :param bool is_express_checkout: Whether the payment is made through express checkout.
+        :return: The compatible payment methods.
+        :rtype: payment.method
+        """
+        compatible_payment_methods = super()._get_compatible_payment_methods(
+            provider_ids, partner_id, currency_id=currency_id, force_tokenization=force_tokenization,
+            is_express_checkout=is_express_checkout)
+        # Each payment method used in Ecuador needs to have the field l10n_ec_sri_payment_id set
+        # (the info is required for invoices when using Ecuadorian EDI)
+        if self.env.company.country_id.code == 'EC':
```

Comment:
It should depend on the actual country you are setting it for, not on the company in the context?  It seems a bit harsh to restrict it at the base, as you could still get an error to go set the payment type on the payment method.

Reply:
I moved the filtering of the payment methods to a different function (website instead of payment). It now depends on the company of the order